scapy/contrib/homeplugav.py

```diff
@@ -476,6 +476,101 @@ class ReadMACMemoryConfirmation(Packet):
                    StrLenField("MACData", b"\x00", length_from=lambda pkt: pkt.MACLen),  # noqa: E501
                    ]
 
+######################################################################
+# Module Operation (for newest chipset like 6410, 7000, 7420 and 7500)
+######################################################################
+
+
+OperationList = {0x0000: "Read",
+                 0x0011: "Write"}
+
+
+class ModuleOperationRequest(Packet):
+    name = "ModuleOperationRequest"
+    fields_desc = [XIntField("reserved", 0),
+                   XByteField("NumOpData", 0x01),
+                   LEShortEnumField("operation", 0x0000, OperationList),
+                   LEShortField("OPDataLength", None),
+                   XIntField("reserved_1", 0),
+                   ConditionalField(LEIntField("SessionID", 0),
+                                    lambda pkt:(0x0011 == pkt.operation)),
+                   ConditionalField(XByteField("ModuleIDX", 0),
+                                    lambda pkt:(0x0011 == pkt.operation)),
+                   LEShortField("ModuleID", 0x7002),
+                   LEShortField("ModuleSubID", 0x0000),
+                   ConditionalField(LEShortField("ReadDataLen", 0x0578),
+                                    lambda pkt:(0x0000 == pkt.operation)),
+                   ConditionalField(LEIntField("ReadOffset", 0x00000000),
+                                    lambda pkt:(0x0000 == pkt.operation)),
+                   ConditionalField(FieldLenField("WriteDataLen", None,
+                                                  count_of="ModuleData",
+                                                  fmt="<H"),
+                                    lambda pkt:(0x0011 == pkt.operation)),
+                   ConditionalField(LEIntField("WriteOffset", 0x00000000),
+                                    lambda pkt:(0x0011 == pkt.operation)),
+                   ConditionalField(StrLenField("ModuleData", b"\x00",
+                                                length_from=lambda pkt: pkt.WriteDataLen),  # noqa: E501
+                                    lambda pkt:(0x0011 == pkt.operation))]
+
+    def post_build(self, p, pay):
+        if self.operation == 0x0000:
+            if self.OPDataLength is None:
+                _len = 18
+                p = p[:7] + struct.pack('h', _len) + p[9:]
```

Comment:
```suggestion
                p = p[:7] + struct.pack('!H', _len) + p[9:]
```
Please stick to network values (=Big endian). https://docs.python.org/2/library/struct.html#byte-order-size-and-alignment

also applies elsewhere

Reply:
Done